### PR TITLE
Reduce clientside evaluation

### PR DESCRIPTION
**Pull request by Scott-Caldwell**
Trying to batch this out into smaller chunks in case I screw something up. This PR eliminates clientside evaluation of EF queries for a number of our repository calls.

### CODE REVIEW

**Comment by JakenVeina** [comment]
Any chance you could briefly summarize the queries affected here, and how the query strategy has changed? I'm having trouble contextualizing what's changed here.

File: Modix.Data/Repositories/EmojiRepository.cs [hunk 1]
```diff
@@ -277,7 +252,8 @@ limit 1
                     )
                     select ""EmojiId"", ""EmojiName"", ""IsAnimated"", ""Uses"", ""Rank"", ""TopUserId"", ""TopUserUses""
                     from stats, user_stats
-                    where { (emoji.Id is null ? @"""EmojiName"" = :EmojiName" : @"""EmojiId"" = :EmojiId") }";
+                    where {emojiClause}";
```
Comment (JakenVeina) [inline Modix.Data/Repositories/EmojiRepository.cs]: Should this go up in the `stats` subquery, before the `GROUP BY` clause?

File: Modix.Data/Repositories/EmojiRepository.cs [hunk 2]
```diff
@@ -277,7 +252,8 @@ limit 1
                     )
```
Comment (JakenVeina) [inline Modix.Data/Repositories/EmojiRepository.cs]: Should the `"Timestamp" >= :StartTimestamp"` clause not be swapped out for `{dateClause}` for both queries?

File: Modix.Data/Repositories/InfractionRepository.cs [hunk 3]
```diff
@@ -199,23 +202,54 @@ public async Task<IReadOnlyCollection<InfractionSummary>> SearchSummariesAsync(I
                 .ToArrayAsync();
 
         /// <inheritdoc />
-        public async Task<IDictionary<InfractionType, int>> GetInfractionCountsAsync(InfractionSearchCriteria searchCriteria)
+        public async Task<IReadOnlyCollection<InfractionCount>> GetInfractionCountsAsync(ulong guildId, ulong subjectId)
```
Comment (JakenVeina) [inline Modix.Data/Repositories/InfractionRepository.cs]: Why are we reducing functionality here?

File: Modix.Data/Repositories/EmojiRepository.cs (hunk 1 above)
Comment (Scott-Caldwell) [reply Modix.Data/Repositories/EmojiRepository.cs]: I don't think so. I think if I do that, then the rank will always just be 1.

File: Modix.Data/Repositories/EmojiRepository.cs (hunk 2 above)
Comment (Scott-Caldwell) [reply Modix.Data/Repositories/EmojiRepository.cs]: Good catch, will fix.

File: Modix.Data/Repositories/InfractionRepository.cs (hunk 3 above)
Comment (Scott-Caldwell) [reply Modix.Data/Repositories/InfractionRepository.cs]: I mean, I *could* try to fit in all the possible scenarios that come with the search criteria, but so far we haven't needed the extra functionality for this query, and I don't foresee this query being used in a wide variety of cases.

**Comment by Scott-Caldwell** [comment]
I can get started on some unit tests for these ASAP. Honestly, it wouldn't be a bad idea to start requiring them on virtually every PR soon now that we've got a good start on the testing libraries.

File: Modix.Data/Repositories/EmojiRepository.cs (hunk 1 above)
Comment (JakenVeina) [reply Modix.Data/Repositories/EmojiRepository.cs]: Ah, yeah. The `Rank` column is what makes that necessary.

**Comment by JakenVeina** [comment]
For the record, as discussed in Discord, gonna go ahead and proceed with this, since the raw queries can't actually BE unit tested.

File: Modix.Data/ModixContext.cs [hunk 4]
```diff
@@ -75,11 +75,16 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
             foreach(var method in onModelCreatingMethods)
                 method.Invoke(null, new [] { modelBuilder });
 
+            modelBuilder.Query<PerChannelMessageCount>();
```
Comment (JakenVeina) [inline Modix.Data/ModixContext.cs]: Can we move these to `OnModelCreating` methods on each model type?

**Comment by Scott-Caldwell** [comment]
Closing until I have time to work on this

**Pull request closed without merging**